dds/DCPS/PrinterValueWriter.h

```diff
@@ -0,0 +1,290 @@
+/*
+ * Distributed under the OpenDDS License.
+ * See: http://www.opendds.org/license.html
+ */
+
+#ifndef OPENDDS_DCPS_PRINTER_VALUE_WRITER_H
+#define OPENDDS_DCPS_PRINTER_VALUE_WRITER_H
+
+#include "ValueWriter.h"
+#include "dcps_export.h"
+#include "Definitions.h"
+
+#include <dds/DdsDcpsCoreTypeSupportImpl.h>
+#include <dds/DdsDcpsTopicC.h>
+
+#include <iosfwd>
+#include <sstream>
+#include <vector>
+
+#if !defined (ACE_LACKS_PRAGMA_ONCE)
+#pragma once
+#endif /* ACE_LACKS_PRAGMA_ONCE */
+
+OPENDDS_BEGIN_VERSIONED_NAMESPACE_DECL
+
+namespace OpenDDS {
+namespace DCPS {
+
+/// Convert values to a readable format.
+class PrinterValueWriter : public ValueWriter {
+public:
+  static unsigned int indent;
```

Comment:
Is this supposed to be `const`? If not, how does it handle concurrency/reentrancy?

Reply:
No, it's a global default so it should only be changed once at start up.

Reply:
How about using `Service_Participant` to keep track of it like other `[common]` configurable values?